yarn-project/circuits.js/src/keys/derivation.ts

```diff
@@ -127,15 +127,12 @@ export function deriveKeys(secretKey: Fr) {
   };
 }
 
-export function computeTaggingSecret(senderCompleteAddress: CompleteAddress, senderIvsk: Fq, recipient: AztecAddress) {
-  const senderPreaddress = computePreaddress(
-    senderCompleteAddress.publicKeys.hash(),
-    senderCompleteAddress.partialAddress,
-  );
+export function computeTaggingSecret(knownAddress: CompleteAddress, ivsk: Fq, externalAddress: AztecAddress) {
```

Comment:
This has been generalized, since the secret is computed the same whether you're the sender or the recipient. The only important point is what complete address + ivsk is known to us.